esp-hal/src/i2c/master/mod.rs

```diff
@@ -934,6 +937,7 @@ fn configure_clock(
 }
 
 /// Peripheral data describing a particular I2C instance.
+#[derive(Debug, Clone, Copy)]
```

Comment:
Not sure if these are all appropriate or not, just kind of guessed based on what the compiler allowed for this one 😅

Reply:
Take 2: IIRC we have a quasi-agreement of not exposing these structs publicly anyway. That makes me think that whatever you do here is fine, and we'll have to return to this once we commit to stabilizing them.

Reply:
true - but still doesn't feel right to have them Copy + Clone 🤷‍♂️